Tela de Balances

TTOExtreme opened the pull request:
-- Adicionado tela /balances
-- Carregamento dinâmico das moedas na tela
-- Alterado tabela para div com flexbox
-- Ao clicar em deposito ira abrir uma janela com o endereço (enviado pelo server) para realizar um deposito
-- Gera um qr code a partir do endereço recebido
-- Colocado um filtro para aceitar somente numeros, e sómente um ponto ou uma virgula

Commit:
Tela de Balances 

-- adicionado rota para /balances
-- criado tela com exemplo de dados
-- deposito e saque 'funcional' (necessita implementar requests no back-end)

Commit:
Tela de Balances 

-- adicionado rota para /balances
-- criado tela com exemplo de dados
-- deposito e saque 'funcional' (necessita implementar requests no back-end)

Commit:
Melhoria de estetica

Commit:
Melhoria de estetica

Commit:
Clear code

-- Retirado codigo inutilzado

Commit:
Clear code

-- Retirado codigo inutilzado

gabr1gus committed:
Protótipo de servidor para o módulo externo da NANO

-- Criado npm module próprio para o módulo nano
-- Collections de accounts é limpa e recriada na inicialização do modulo
-- Ao iniciar, o servidor irá fazer o comando GET account_list/nano para recuperar a lista de accounts NANO atualizada do servidor principal
-- Criação de funções baseadas nos comandos RPC
-- Criação de um servidor que ira lidar com comandos POST e GET direcionados a API da NANO
-- Adição de um client para o websocket da NANO, que verifica as operações realizadas na rede
-- Adicionado env para o mongodb, porta do servidor e url do servidor principal

GCastilho committed:
Implementado create_account na currencyApi (#16)

Geral:
-- A página de cadastro agora requisita uma account nano da CurrencyApi ao salvar o usuário no banco de dados
-- Currencies agora tem uma propriedade 'ip', que indica o IP da máquina em que seu módulo externo está rodando
-- A propriedade que identifica o nome da currency nos módulos individuais agora se chama 'name' e não 'currency'
-- A propriedade 'name' de uma currency na CurrencyApi agora é colocado automaticamente pelo index da currencies, não sendo necessário ter uma propriedade para isso

Módulos individuais:
-- Implementado validador de nano accounts ao salvar um usuário no database
-- Porta da NANO configurada para 8090

Listener:
-- Refatorado módulo listener para separar funções acessíveis pelas APIs externas através do listener das funções internas da CurrencyApi
-- As funções acessíveis externamente devem ficar no pseudo-módulo 'functions', com cada função sendo um sub-módulo diferente
-- As funções do módulo 'commons' agora só são acessíveis internamente, não estando disponíveis para serem acessíveis pelo listener

Bugs:
-- Corrigido problema de o cache do require do módulo das currencies estar sendo modificado (globalmente) na instanciação da CurrencyApi

Testes:
-- Adicionado mock (manual) na CurrencyApi para não chamar o módulo externo ao requisitar uma account NANO nova nos testes
-- Adicionado testes para garantir que as funções do listener são acessíveis e funções individuais ou da commons não
-- Adicionado testes no listener para comando inexistente e currency inexistente

GCastilho reviewed (changes requested):
O código não compila :(

> Module not found: Can't resolve '../../components/BalancesTableItem/BalancesTableItem' in '[...]/unip-tcc/client/src/pages/balances'

joaojvds committed:
Implementado layout inicial da pagina home e o unit tests (#14)

-- Colocado as cores do site em variáveis
-- Separado mensagem de erro em componente
-- Criado páginas main e not found
-- Configurado rotas no arquivo router
-- Alterado aparência do header
-- Alterado aparência das paginas de login e cadastro
-- Consertado redirect da pagina login, ele agora não recarrega o site
-- O navbar agora se adapta a qualquer tamanho de tela
-- Adicionado um drawer menu para a navegação no mobile
-- Adicionado unit tests no front end
-- Removido a propriedade 'httpOnly' do cookie enviado pelo servidor
-- Botão logout só aparece quando há um cookie de sessão
-- Refatorado funções dos arquivos login, register e header
-- Adicionado redirect para a home se estiver logado nas paginas login e register (não é mais possível acessá-las quando estiver logado)
-- Botão logout faz o redirect para a home após deslogá-lo
-- Ao fazer login, o cliente verifica se um cookie foi recebido, disparando uma mensagem de erro caso contrário 
-- Alterado cor do input

joaojvds committed:
Alterado estrutura da tabela

-- Alterado tabela para div com flexbox

joaojvds committed:
Merge remote-tracking branch 'origin/Balances-Page' into Balances-Page

joaojvds committed:
Consertado cor dos números da tabela

-- Consertado cor dos números da tabela, que não estavam aparecendo

joaojvds committed:
Janela de deposito

-- Ao clicar em deposito ira abrir uma janela com o endereço (enviado pelo server) para realizar um deposito
-- Gera um qr code a partir do endereço recebido

joaojvds committed:
Desing basico da aba de saque

-- Desing basico ainda em desenvolvimento

joaojvds committed:
Alteração do css 

-- Alterando o estilo da aba

joaojvds committed:
Aba de withdraw

-- Terminado css da aba
-- Consertado bug ao abrir outra aba
-- Criado handle para os input
-- Colocado um filtro para aceitar somente numeros, e sómente um ponto ou uma virgula

joaojvds committed:
Remove unnecessary escape

-- Removido warnings das expressões regulares

GCastilho reviewed (approved):
Da próxima vez fz um rebase ao invés de um merge
